operations/mimir/memcached.libsonnet

```diff
@@ -50,48 +50,52 @@ memcached {
   },
 
   // Dedicated memcached instance used to cache query results.
-  memcached_frontend: if $._config.memcached_frontend_enabled then
-    $.memcached {
-      name: 'memcached-frontend',
-      max_item_size: '%dm' % [$._config.memcached_frontend_max_item_size_mb],
-      connection_limit: 16384,
-    } + if $._config.memcached_frontend_mtls_enabled then $.memcached_mtls else {}
-  else {},
+  memcached_frontend:
+    if $._config.cache_frontend_enabled && $._config.cache_frontend_backend == 'memcached' then
+      $.memcached {
+        name: $._config.cache_frontend_name,
+        max_item_size: '%dm' % [$._config.cache_frontend_max_item_size_mb],
+        connection_limit: 16384,
+      } + if $._config.memcached_frontend_mtls_enabled then $.memcached_mtls else {}
+    else {},
 
   // Dedicated memcached instance used to temporarily cache index lookups.
-  memcached_index_queries: if $._config.memcached_index_queries_enabled then
-    $.memcached {
-      name: 'memcached-index-queries',
-      max_item_size: '%dm' % [$._config.memcached_index_queries_max_item_size_mb],
-      connection_limit: 16384,
-    } + if $._config.memcached_index_queries_mtls_enabled then $.memcached_mtls else {}
-  else {},
+  memcached_index_queries:
+    if $._config.cache_index_queries_enabled && $._config.cache_index_queries_backend == 'memcached' then
+      $.memcached {
+        name: $._config.cache_index_queries_name,
```

Comment:
The name doesn't need to be configurable. It's `memcached-*` for Memcached and `redis-*` for Redis.

Reply:
We use this name for consistent URL building for both redis and memcached. I.e.
```
// memcached
'query-frontend.results-cache.memcached.addresses': 'dnssrvnoa+%(cache_frontend_name)s.%(namespace)s.svc.cluster.local:11211' % $._config,
// redis
'blocks-storage.bucket-store.results-cache.redis.endpoint': '%(cache_frontend_name)s.%(namespace)s.svc.cluster.local:6379' % $._config,
```

It is easier to have the name extracted as a variable in the config rather than have everything hard coded.

Reply:
In that case it should be constructed from the cache backend variable. There shouldn't be _two_ settings that need to be changed when switching between Memcached and Redis